app/src/main/java/org/schabi/newpipe/util/SaveUploaderUrlHelper.java

```diff
@@ -0,0 +1,81 @@
+package org.schabi.newpipe.util;
+
+import static org.schabi.newpipe.extractor.utils.Utils.isNullOrEmpty;
+
+import android.content.Context;
+import android.widget.Toast;
+
+import androidx.fragment.app.Fragment;
+
+import org.schabi.newpipe.NewPipeDatabase;
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.error.ErrorInfo;
+import org.schabi.newpipe.error.ErrorUtil;
+import org.schabi.newpipe.error.UserAction;
+import org.schabi.newpipe.extractor.stream.StreamInfoItem;
+import org.schabi.newpipe.player.playqueue.PlayQueueItem;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+/**
+ * saveUploaderUrlIfNeeded will check if the uploaderUrl is present in the StreamInfoItem
+ * or PlayQueueItem, and if so will immediately call the callback with the uploaderUrl.
+ * Otherwise, it will use the ExtractorHelper to get the uploaderUrl and save it to the
+ * database for future use, then call the callback with the uploaderUrl.
+ */
```

Comment:
I'm not very good at writing documentation, can someone please help to reword this if needed?

Reply:
I would have gone with something simple like:
``Utility class for putting the uploader url into the database - when required.``

Reply:
Fixed in https://github.com/TeamNewPipe/NewPipe/pull/7538/commits/607d9959faaa440ceb83e4e57a133d577223da3f